project_task_code/models/project_task.py

```diff
@@ -28,3 +29,22 @@ def copy(self, default=None):
             default = {}
         default['code'] = self.env['ir.sequence'].next_by_code('project.task')
         return super(ProjectTask, self).copy(default)
+
+    @api.model
+    def name_search(self, name='', args=None, operator='ilike', limit=100):
+        res = super(ProjectTask, self).name_search(
```

Comment:
it's quite wasteful to always do two searches. I think the logic of this function should be something like

```python
result = None
if name.looks_like_code():
    result = our_search()
if not result:
    result = super()
return result
```

This way we usually only do one search, and only when something that looks like a code but isn't a code (and that should be the least usual case) is filled in, do both.